ci-operator/step-registry/cucushift/installer/check/ibmcloud/resourcegroup/cucushift-installer-check-ibmcloud-resourcegroup-commands.sh

```diff
@@ -0,0 +1,107 @@
+}
+
+#OCP-60816 - [IPI-on-IBMCloud] install cluster under BYON with a different resource group
+function getResources() {
+    "${IBMCLOUD_CLI}" resource service-instances --type all --output JSON | jq -r '.[]|.name+" "+.resource_id' | sort
+}
+
+function checkResourcesInVPC() {
+    local preResourceFile resource_group curResourceFile cmpRst
+    resource_group=$1
+    preResourceFile="${SHARED_DIR}/vpc_resources"
+    "${IBMCLOUD_CLI}" target -g ${resource_group}
+    curResourceFile=$(mktemp)
+    getResources > ${curResourceFile}
+    echo "diff ${curResourceFile} ${preResourceFile} ..."
+    cmpRst=$(diff ${curResourceFile} ${preResourceFile})
+    if [ -n "$cmpRst" ]; then
+        echo "ERROR: resources differ: ${cmpRst}"
+        return 1
+    else
+        echo "checkResourcesinVPC PASS"  
+        return 0
+    fi
+}
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+else
+    echo "ERROR: fail to get the kubeconfig file under ${SHARED_DIR}!!"
+    exit 1
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+check_result=0
+ibmcloud_login
+
+provisioned_rg_file="${SHARED_DIR}/ibmcloud_resource_group"
+cluster_resource_group=$(jq -r .ibmcloud.resourceGroupName ${SHARED_DIR}/metadata.json)
+rg_file="${SHARED_DIR}/ibmcloud_cluster_resource_group"
+if [ -f "${provisioned_rg_file}" ]; then
+    vpc_resource_group=$(cat "${provisioned_rg_file}")
+fi
+
+echo "CREATE_CLUSTER_RESOURCE_GROUP: ${CREATE_CLUSTER_RESOURCE_GROUP}"
+
+# check the cluster_resource_group saved in the files.
+if [ "${CREATE_CLUSTER_RESOURCE_GROUP}" == "yes" ]; then
+    if [ -s "${rg_file}" ]; then            
+        if [ "${vpc_resource_group}" == "${cluster_resource_group}" ]; then
+            echo "ERROR: vpc and cluster use the same resource group ${cluster_resource_group} when CREATE_CLUSTER_RESOURCE_GROUP is set!"
+            check_result=1
+        fi
+        rgName=$(cat "${rg_file}")
+        if [ "${rgName}" != "${cluster_resource_group}" ]; then
+            echo "ERROR: in ${rg_file} saved ${rgName} is not same with the cluster name ${cluster_resource_group}!"
+            check_result=1
+        fi
+
+        if ! checkResourcesInVPC ${vpc_resource_group}; then
+            check_result=1
+        fi
+
+    else
+        echo "ERROR: fail to find ${rg_file} when CREATE_CLUSTER_RESOURCE_GROUP is set!"
+        check_result=1
+    fi
+else
+    if [ -s "${rg_file}" ]; then
+        echo "ERROR: find the ${rg_file} when CREATE_CLUSTER_RESOURCE_GROUP is not YES!"
+        check_result=1
+    fi
+
+    if [ -s ${provisioned_rg_file} ]; then
+        rgName=$(cat "${provisioned_rg_file}")
+        if [ "${rgName}" != "${cluster_resource_group}" ]; then
+            echo "ERROR: in ${provisioned_rg_file} saved ${rgName} is not same with the cluster name ${cluster_resource_group}!"
+            check_result=1
+        fi
+    fi
+fi
```

Comment:
I did not the change per my last comment.

Reply:
Can you recheck my proposed code in https://github.com/openshift/release/pull/51310#discussion_r1590490184 ? Does it not meet your test requirement?